pkg/controller/cassandradatacenter/metadata.go

```diff
@@ -20,6 +20,16 @@ func DataCenterLabels(cdc *cassandraoperatorv1alpha1.CassandraDataCenter) map[st
 	}
 }
 
+func PrometheusLabels(cdc *cassandraoperatorv1alpha1.CassandraDataCenter) map[string]string {
+	// Fetch cdc labels
+	labels := DataCenterLabels(cdc)
+	// Add prometheus labels if defined
+	for label, val := range cdc.Spec.PrometheusServiceMonitorLabels {
+		labels[label] = val
```

Comment:
do we really need to propagate all labels from `DataCenterLabels(cdc)` to Prometheus operator? Isn't just an empty map to start with fine?

Reply:
Yes, we do need the CDC labels here. The Prometheus service that we create (`cassandra-<cdc name>-prometheus`, which is where it is used) selects pods via labels, and that's where we need those labels - in the service selector. Then the `ServiceMonitor` has a selector to pick the service for monitoring, and it will use the same set of labels to pick a service for the specific CDC; then the Prometheus resource (which we don't care for nor create) will use its own selector to find ServiceMonitors.

We don't propagate our CDC labels to Prometheus operator. These labels are for our internal use and for the `ServiceMonitor` that we create.